### DISCUSSION TITLE
Default html template for custom email?

### DISCUSSION THREAD

**Discussion by AugustBurnsRed** [strapi/strapi#6434] [Legacy Topics]
Hi, i was wondering if there is a way to make a default html template for custom email? In the controller file i send mail programmatically like https://strapi.io/documentation/3.0.0-beta.x/guides/send-email.html#introduction

Can we add an other template like here /users-permissions/email-templates and then reuse it when we send our email? I would not want to always put the header and footer in 'html:'. Thats seem pretty basic but i can't find any exemple.

Thanks!

**Comment by lauriejim** [comment]
Hello! Here is the code that add the default template in the Users-Permissions section - https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/config/functions/bootstrap.js#L110

If you want add your custom template there, you will have to create a script that:
- get the current templates
```js
const pluginStore = strapi.store({
  environment: '',
  type: 'plugin',
  name: 'users-permissions',
});

const templates = await pluginStore.get({ key: 'email' }));
```
- the you are adding yours by following the correct structure
```js
templates.custom = {
  display: 'Email.template.reset_password',
  icon: 'sync',
  options: {
   from: {
     name: 'Administration Panel',
     email: 'no-reply@strapi.io',
   },
   response_email: '',
   object: 'Reset password',
   message: `<p>We heard that you lost your password. Sorry about that!</p>
<p>But don’t worry! You can use the following link to reset your password:</p>
<p><%= URL %>?code=<%= TOKEN %></p>
<p>Thanks.</p>`,
  },
},
```
- and finally you save the new template
```js
await pluginStore.set({ key: 'email', templates});
```
You can run this login in the `config/functions/bootstrap.js` file.
Then start your server on time.
And then remove the code.
Restart your app and make sure your template has been added.

**Comment by AugustBurnsRed** [comment]
Thanks! I don't know if i'm doing something wrong but with your code it's not working (error TypeError: Cannot read property 'toString' of undefined) when i try to do 'await pluginStore.set({ key: 'email', templates});' I'm on v3.0.2

But instead i just took the code from https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/config/functions/bootstrap.js#L110 , then added my custom email and it worked. I just have to make sure that if i modify the default template, i have to modify it too in my bootstrap if i decide to run an other custom template.

To get the template and modify it in my controller, i used the code from https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js#L305

**Edit**
After adding a new template, there is a error when i try to modify any template. (debug PUT /users-permissions/email-templates (5 ms) 400)

**Reply by EdwardKerckhof** [reply]
Have you figured out a way to get past the PUT Bad Request error?

**Comment by alanhr** [comment]
Hello, @lauriejim .

Why Strapi not create a mail template table like it has in image?
